spec/models/users/notification_setting_spec.rb

```diff
@@ -0,0 +1,44 @@
+require "rails_helper"
+
+RSpec.describe Users::NotificationSetting, type: :model do
+  let!(:user) { create(:user) }
+  let(:notification_setting) { described_class.find_by(user_id: user.id) }
+
+  describe "validations" do
+    subject { notification_setting }
+
+    describe "builtin validations" do
+      it { is_expected.to validate_inclusion_of(:email_digest_periodic).in_array([true, false]) }
+    end
+  end
+
+  context "when callbacks are triggered after commit" do
+    describe "subscribing to mailchimp newsletter" do
+      it "enqueues SubscribeToMailchimpNewsletterWorker when updating email_newsletter to true" do
+        sidekiq_assert_enqueued_with(job: Users::SubscribeToMailchimpNewsletterWorker, args: [user.id]) do
+          notification_setting.update(email_newsletter: true)
+        end
+      end
+
+      it "enqueues SubscribeToMailchimpNewsletterWorker when updating email_newsletter to false" do
+        notification_setting.update(email_newsletter: true)
+        sidekiq_assert_enqueued_jobs(1, only: Users::SubscribeToMailchimpNewsletterWorker) do
```

Comment:
I used not `sidekiq_assert_enqueued_with` but `sidekiq_assert_enqueued_jobs` for checking the jobs count. Because L24 pushes a `Users::SubscribeToMailchimpNewsletterWorker` job as well.